modules/schema/creation_test.go

```diff
@@ -6,148 +6,1379 @@ import (
+			args: args{
+				dbAlias:       "postgres",
+				tableName:     "table1",
+				project:       "test",
+				parsedSchema:  schemaType{"postgres": schemaCollection{"table1": SchemaFields{"col1": &SchemaFieldType{FieldName: "col1", Kind: TypeID, IsFieldTypeRequired: true, IsPrimary: true}}}},
+				currentSchema: schemaCollection{"table1": SchemaFields{"col1": &SchemaFieldType{FieldName: "col1", Kind: TypeID, IsPrimary: false}}},
+			},
+			fields:  fields{crud: crudPostgres, project: "test"},
+			want:    []string{"ALTER TABLE test.table1 ALTER COLUMN col1 SET NOT NULL", "ALTER TABLE test.table1 ADD CONSTRAINT c_table1_col1 PRIMARY KEY (col1)"},
+			wantErr: false,
+		},
+		{
+			name: "removing primary key",
+			args: args{
+				dbAlias:       "postgres",
+				tableName:     "table1",
+				project:       "test",
+				parsedSchema:  schemaType{"postgres": schemaCollection{"table1": SchemaFields{"col1": &SchemaFieldType{FieldName: "col1", Kind: TypeID, IsFieldTypeRequired: false, IsPrimary: false}}}},
+				currentSchema: schemaCollection{"table1": SchemaFields{"col1": &SchemaFieldType{FieldName: "col1", Kind: TypeID, IsFieldTypeRequired: true, IsPrimary: true}}},
+			},
+			fields:  fields{crud: crudPostgres, project: "test"},
+			want:    []string{"ALTER TABLE test.table1 ALTER COLUMN col1 DROP NOT NULL", "ALTER TABLE test.table1 DROP CONSTRAINT c_table1_col1"},
+			wantErr: false,
+		},
+		{
+			name: "adding foreign key with type change",
+			args: args{
+				dbAlias:       "postgres",
+				tableName:     "table1",
+				project:       "test",
+				parsedSchema:  schemaType{"postgres": schemaCollection{"table1": SchemaFields{"col1": &SchemaFieldType{FieldName: "col1", Kind: typeInteger, IsForeign: true, JointTable: &TableProperties{Table: "table2", To: "id"}}}}},
+				currentSchema: schemaCollection{"table1": SchemaFields{"col1": &SchemaFieldType{FieldName: "col1", Kind: TypeID, IsForeign: false}}, "table2": SchemaFields{}},
+			},
+			fields:  fields{crud: crudPostgres, project: "test"},
+			want:    []string{"ALTER TABLE test.table1 DROP COLUMN col1", "ALTER TABLE test.table1 ADD COLUMN col1 bigint", "ALTER TABLE test.table1 ADD CONSTRAINT c_table1_col1 FOREIGN KEY (col1) REFERENCES test.table2 (id)"},
+			wantErr: false,
+		},
+		{
+			name: "removing foreign key",
+			args: args{
+				dbAlias:       "postgres",
+				tableName:     "table1",
+				project:       "test",
+				parsedSchema:  schemaType{"postgres": schemaCollection{"table1": SchemaFields{"col1": &SchemaFieldType{FieldName: "col1", Kind: TypeID, IsFieldTypeRequired: false, IsForeign: false}}, "table2": SchemaFields{"id": &SchemaFieldType{FieldName: "col1", Kind: TypeID}}}},
+				currentSchema: schemaCollection{"table1": SchemaFields{"col1": &SchemaFieldType{FieldName: "col1", Kind: TypeID, IsForeign: true, JointTable: &TableProperties{Table: "table2", To: "id"}}}},
 			},
-			Enabled: true,
+			fields:  fields{crud: crudPostgres, project: "test"},
+			want:    []string{"ALTE
```

Comment:
@YourTechBud does the return statement here if an error occurs will stop testing the other test cases?

Reply:
There should be no return statements here